metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -894,6 +895,27 @@ def _make_select_columns_for_metrics(
                 else:
                     select_expression = column_reference_expression
 
+                # At this point, the MetricSpec might have the alias in place of the element name, so we need to look
+                # back at where it was defined from to get the metric element name.
+                metric_name = metric_instance.defined_from.metric_name
+                input_measures = self._metric_lookup.measures_for_metric(
+                    metric_reference=MetricReference(metric_name),
+                    column_association_resolver=self._column_association_resolver,
+                )
+                # If multiple input measures, this is a query with a nested derived/ratio metric. In that case, we can
+                # skip this step because it has already occurred when rendering the nested metric.
+                # TODO: this logic might need updating for conversion metrics, which will allow multiple input measures
```

Comment:
Can we make this condition based on metric type (or some is_derived property or similar) instead of number of measures?